src/Lender.sol

```diff
@@ -512,29 +512,81 @@ contract Lender {
         uint8 p,
         address u,
         uint256 m,
+        uint256[] memory a,
         bytes calldata d
     ) external returns (uint256) {
         IMarketPlace.Market memory _Market = IMarketPlace(marketPlace).markets(u, m);
 
-        // Fetch the adapter for this lend call
-        address adapter = _Market.adapters[p];
+        // Conduct the lend operation to acquire principal tokens
+        (bool success, bytes memory returndata) = _Market.adapters[p].delegatecall(
+            abi.encodeWithSignature('lend(uint256[] amount, bool internalBalance, bytes calldata inputdata)', a, false, d));
 
-        // Fetch the principal token for this lend call
-        address pt = _Market.tokens[p];
+        if (!success) {
+            revert Exception(0, 0, 0, address(0), address(0)); // TODO: assign exception
+        }
+
+        // Get the amount of PTs (in protocol decimals) received
+        (uint256 obtained, uint256 spent) = abi.decode(
+            returndata,
+            (uint256, uint256)
+        );
+
+        // Convert decimals from principal token to underlying
+        uint256 returned = convertDecimals(u, _Market.tokens[p], obtained);
+
+        // Mint Illuminate PTs to msg.sender
+        IERC5095(principalToken(u, m)).authMint(msg.sender, returned);
+        emit Lend(p, u, m, returned, spent, msg.sender);
+        return (returned);
+    }
+
+    // An override lend method for all ETH lending with the additional lpt and swap parameters
+    // This method is only used for lending to ETH markets
+    function lend(
```

Comment:
I would continue to call it `lendETH` or similar... in my mind, if we don't pass `lst` and `swapMinimum`, the function should behave like the function above, right? since it's a different function used only for lending to ETH markets, I would change it name 🤷‍♂️

Reply:
Yeah ill leave that up to @alexanderwende -- If he likes the overrides we'll keep them, otherwise we can name it more specifically

Reply:
I don't mind either way to be honest... in the JS world it's pretty common to have a function that behaves differently based on the input params. so this doesn't make me uncomfortable. in the end, you're lending, so calling the method `lend` feels correct to me and i don't have to remember that there are 2 different functions in order to lend...

i'd prefer relying on my IDE to tell me that there are different signatures available when calling that method in code.

on the other hand, having an explicit `lendETH` method doesn't feel horrible either... 🤷